scripts/src/javatests/org/oppia/android/scripts/coverage/CoverageRunnerTest.kt

```diff
@@ -40,89 +82,132 @@ class CoverageRunnerTest {
   }
 
   @Test
-  fun testRunWithCoverageAsync_emptyDirectory_throwsException() {
+  fun testRunCoverageForTestTarget_emptyDirectory_throwsException() {
     val exception = assertThrows<IllegalStateException>() {
-      runBlocking {
-        coverageRunner.runWithCoverageAsync(bazelTestTarget).await()
-      }
+      coverageRunner.runCoverageForTestTarget(bazelTestTarget)
     }
 
     assertThat(exception).hasMessageThat().contains("not invoked from within a workspace")
   }
 
   @Test
-  fun testRunWithCoverageAsync_invalidTestTarget_throwsException() {
+  fun testRunCoverageForTestTarget_invalidTestTarget_throwsException() {
     testBazelWorkspace.initEmptyWorkspace()
 
     val exception = assertThrows<IllegalStateException>() {
-      runBlocking {
-        coverageRunner.runWithCoverageAsync(bazelTestTarget).await()
-      }
+      coverageRunner.runCoverageForTestTarget(bazelTestTarget)
     }
 
     assertThat(exception).hasMessageThat().contains("Expected non-zero exit code")
     assertThat(exception).hasMessageThat().contains("no such package")
   }
 
   @Test
-  fun testRunWithCoverageAsync_validSampleTestTarget_returnsCoverageData() {
+  fun testRunCoverageForTestTarget_coverageRetrievalFailed_throwsException() {
```

Comment:
I suppose that this is still not complete? re - commented out code.

Reply:
It was passing earlier but ever since removing the async flow it is throwing an error saying - 
```
java.io.FileNotFoundException: /home/rddev/.cache/bazel/.../bazel-out/k8-fastbuild/testlogs/
coverage/test/java/com/example/AddNumsTest/coverage.dat (Permission denied)
```
I rechecked it by removing the asynchronous call (again separately) and it seems just removing async calls is causing this (not sure why though), I am still investigating, will update once I could find a fix.

Reply:
@adhiamboperes, the test cases are now fixed by,
1. Replacing the hard coded path with regex to acquire the coverage.dat file to ignore file not found exceptions.
2. Creating the file to make sure it has permissions before over writing the file to simulate the error. 
3. Replicating the actual behaviour for retrieval failure with incorrect package structure `coverage/example`.

Reply:
Can this be verified as completed now @adhiamboperes?

Reply:
This looks resolved, although the test name has since changed.